### DISCUSSION TITLE
Fapolicyd coming features

### DISCUSSION THREAD

**Discussion by radosroka** [ctc-oss/fapolicy-analyzer#778] [Developers]
We(fapolicyd upstream) were discussing two new features in fapolicyd:

Introducing shell escaping for fapolicyd attributes written to syslog to limit tty escape sequences.
E.g.: ` path=/home/rsroka/my\ ls `

Dropping custom escaping from trust file path, it was introduced because space was chosen to be a delimiter in a file and it was not possible to allow file with a space in the name. Since the path is the first item we can read lines backwards `-> hash (space) size (space) name of the file`. Therefore there can be any character in the path of the file.

There is another one which is already there https://github.com/linux-application-whitelisting/fapolicyd/commit/ee9c99a8ac9a13e42a97979fe774b6a412c8d9c6 about passing matched rule number to audit. This is the kernel patch https://lore.kernel.org/all/cover.1675373475.git.rgb@redhat.com/ which provides functionality.

**Comment by jw3** [comment]
Thanks for the heads up on these @radosroka  :+1: 

> Introducing shell escaping for fapolicyd attributes written to syslog to limit tty escape sequences.

Handling the escapes will require some adjustment in the syslog parser.

Added an issue, #781 to track this

> Dropping custom escaping from trust file path, it was introduced because space was chosen to be a delimiter in a file and it was not possible to allow file with a space in the name. Since the path is the first item we can read lines backwards -> hash (space) size (space) name of the file. Therefore there can be any character in the path of the file.

The trust records are being parsed with reverse split, so we should be good there.  I'll add a unit test to track that.

> There is another one which is already there https://github.com/linux-application-whitelisting/fapolicyd/commit/ee9c99a8ac9a13e42a97979fe774b6a412c8d9c6 about passing matched rule number to audit.

That one is being tracked over here in #294
